### PR TITLE
[13.0] [ADD] sale_gathering: Add new module sale_gathering

### PR DESCRIPTION
**Pull request by nicomacr**

### CODE REVIEW

File: sale_gathering/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    _inherit = 'sale.order'
+
+    is_gathering = fields.Boolean('Is Gathering?')
+
+    amount_gathering = fields.Float(
+        compute="_compute_amount_gathering",
+        digits='Product Price',
+        )
+
+    @api.depends('order_line', 'is_gathering')
```
Comment (jjscarafia) [inline sale_gathering/models/sale_order.py]: entiendo que deberiamos cambiar el depends a:
```
    @api.depends(
        'is_gathering'
        'order_line.price_unit_with_tax',
        'order_line.qty_invoiced',
        'order_line.is_downpayment',
     )

```

File: sale_gathering/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    is_gathering = fields.Boolean('Is Gathering?')
+
+    amount_gathering = fields.Float(
```
Comment (jjscarafia) [inline sale_gathering/models/sale_order.py]: renombraría a  "Saldo Acopio" (gathering_balance)

File: sale_gathering/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                if final:
+                    invoiceable_lines |= line
+        return invoiceable_lines
+
+    @api.constrains('is_gathering', 'amount_total')
+    def _check_amount_gathering(self):
+        product_precision_digits = self.env['decimal.precision'].precision_get('Product Price')
+        for rec in self.filtered('is_gathering'):
+            if float_compare(rec.amount_gathering, 0.0, precision_digits=product_precision_digits) == -1:
+                raise ValidationError(_("The amount gathering will be negative, you cannot add more products"
```
Comment (jjscarafia) [inline sale_gathering/models/sale_order.py]: Dejaría así el mensaje

"The gathering balance will be negative (%s), you cannot make this modification to the order. Order: %s" % (rec.amount_gathering, rec.name)

**Review by jjscarafia — CHANGES_REQUESTED** [review]
@nicomacr un par de pavadas..

**Comment by roboadhoc** [comment]
Sorry, I didn't know about this PR and had to retrieve its information, you may have to re-approve it.

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://runbot.adhoc.com.ar/ingadhoc/sale/pull/432).

**Comment by jjscarafia** [comment]
@roboadhoc r+

**Comment by jjscarafia** [comment]
@roboadhoc rebase-ff

**Comment by roboadhoc** [comment]
Merge method set to rebase and fast-forward

**Pull request closed without merging**